### ISSUE TITLE
add rustdocs to CI and publish to github pages

### ISSUE THREAD

**Issue by zimirza** [hd-gmbh-dev/quick-microservice-rs#11]

**Comment by michewl** [comment]
I don't think that publishing the docs to GitHub Pages has any benefit over [docs.rs](https://docs.rs/qm/latest/qm/).

We probably should use the GitHub Pages to publish some end-user targeted documentation and/or detailed concept description for our feature set.

**Comment by kilork** [comment]
I remember idea was to have documentation for trunk version before publishing to crates.io

**Comment by michewl** [comment]
> I remember idea was to have documentation for trunk version before publishing to crates.io

Is it necessary to have the latest? If someone want's to use unreleased features, they must use the repository anyway, meaning they can build the docs locally.

**Comment by kilork** [comment]
@michewl I do not have strong opinion on this. I would merge related PR already, if it would be completely moved to separate file, as this only one thing I think is important. We anyway do not have a book or documentation on ghpages, any documentation then is better than nothing.

**Comment by kilork** [comment]
Not relevant at the moment.
